src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,395 @@
+    'rounded-md',
+    'active:opacity-80',
+    'focus-visible:ring-offset-2',
+    'focus:ring-primary-600',
+    'focus:ring-2',
+    'focus:ring-offset-2',
+    'cursor-pointer',
+  ],
+  {
+    variants: {
+      variant: {
+        default: ['border-primary-600', 'hover:bg-primary-200'],
+        outline: '',
+      },
+      disabled: {
+        true: '',
+        false: '',
+      },
+      error: {
+        true: '',
+        false: '',
+      },
+    },
+    compoundVariants: [
+      {
+        variant: 'default',
+        disabled: true,
+        className: [
+          'border-gray-400',
+          'text-gray-400',
+          'active:opacity-100',
+          'disabled:pointer-events-none',
+          'hover:bg-transparent',
+          'cursor-default',
+        ],
+      },
+      {
+        variant: 'default',
+        error: true,
+        className: [
+          'text-red-700',
+          'ring-1',
+          'ring-red-600',
+          'hover:ring-red-800',
+          'focus:outline-red-600',
+        ],
+      },
+    ],
+    defaultVariants: {
+      variant: 'default',
+    },
+  },
+);
+
+const optionsColor = {
+  default: 'text-primary-500 hover:text-primary-800',
+  red: 'text-red-500 hover:text-red-900',
+  green: 'text-green-500 hover:text-green-800',
+};
+
+interface OptionsBaseProps {
+  color?: 'red' | 'green' | 'default';
+}
+
+export interface OptionsType extends OptionsBaseProps {
+  id: string;
+  label: React.ReactNode;
+  description?: React.ReactNode;
+  disabled?: boolean;
+  category?: never;
+}
+
+interface CategoryOptionType extends OptionsBaseProps {
+  id?: never;
+  label?: never;
+  category: string;
+  disabled?: never;
+  description?: never;
+}
+
+export type OptionsProps = OptionsType | CategoryOptionType;
+
+interface DropdownProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    VariantProps<typeof dropdownVariants> {
+  asChild?: boolean;
+  id: string;
+  placeholder: string;
+  disabled?: boolean;
+  value?: OptionsType;
+  options: OptionsProps[];
+  label?: string;
+  helperText?: string;
+  error?: boolean;
+  errorMessage?: string;
+  optionsContainerHeight?: string;
+  dropDownWidth?: string;
+  onValueChange?: (value: OptionsType) => void;
+}
+
+const Dropdown = React.forwardRef<HTMLDivElement, DropdownProps>(
+  (
+    {
+      className,
+      variant,
+      id,
+      placeholder,
+      options,
+      onValueChange,
+      value,
+      label,
+      errorMessage,
+      error = false,
+      helperText,
+      optionsContainerHeight,
+      dropDownWidth,
+      disabled = false,
+      asChild = false,
+      ...props
+    },
+    ref,
+  ) => {
+    const Comp = asChild ? Slot : 'div';
+
+    const dropdownRef = useRef<HTMLDivElement>(null);
+    const optionRefs = useRef<(HTMLDivElement | null)[]>([]);
+
+    const [isOpen, setIsOpen] = useState(false);
+    const [clickEnabled, setClickEnabled] = useState(true);
+    const [selectedId, setSelectedId] = useState<string>();
```

Comment:
Just use `useState('')` instead, just like you to for clickEnabled above

Reply:
I will continue to do like you suggested. This "<string>" is a old habit.